pallets/oracle-collection/src/lib.rs

```diff
@@ -319,20 +422,92 @@ pub mod pallet {
 
 		/// Sets a associated information to a collection.
 		#[pallet::weight(T::WeightInfo::set_collection_info())]
-		#[pallet::call_index(3)]
-		pub fn set_collection_info(
+		#[pallet::call_index(5)]
+		pub fn create_collection(
+			origin: OriginFor<T>,
+			collection_id: T::CollectionId,
+			feeders: BoundedBTreeSet<T::FeederId, T::MaxFeeders>,
+		) -> DispatchResult {
+			let who = ensure_signed(origin)?;
+
+			ensure!(
+				!CollectionInfo::<T>::contains_key(collection_id),
+				Error::<T>::CollectionAlreadyCreated
+			);
+
+			ensure!(
+				T::IsEditor::check(Edit::new(
+					who,
+					collection_id,
+					Change::CreateCollection(collection_id)
+				)),
+				Error::<T>::IsNotEditor
+			);
+
+			CollectionInfo::<T>::insert(
+				collection_id,
+				CollectionInfo {
+					max_lifetime: None,
+					feeders,
+				},
+			);
+
+			Ok(())
+		}
+
+		/// Proposes a key to be used in the collection
+		#[pallet::weight(T::WeightInfo::set_collection_info())]
+		#[pallet::call_index(6)]
+		pub fn add_key(
 			origin: OriginFor<T>,
 			collection_id: T::CollectionId,
-			info: types::CollectionInfo<T>,
+			key: T::OracleKey,
+			feeders: BoundedVec<FeederInfo<T>, T::MaxFeedersPerKey>,
+			min_feeders: Option<u32>,
 		) -> DispatchResult {
 			let who = ensure_signed(origin)?;
 
 			ensure!(
-				T::IsAdmin::check((who, collection_id)),
-				Error::<T>::IsNotAdmin
+				T::IsEditor::check(Edit::new(
+					who,
+					collection_id,
+					Change::AddKey(
+						collection_id,
+						key,
+						feeders
+							.clone()
+							.try_into()
+							.expect("Size of vec matches size of set at most. qed.")
+					)
+				)),
+				Error::<T>::IsNotEditor
+			);
+
+			CollectionInfo::<T>::try_mutate(&collection_id, |info| {
+				let info = info.as_mut()?;
+
+				for feeder in &feeders {
+					ensure!(info.feeders.contains(feeder), Error::<T>::FeederNotAllowed);
+					Self::maybe_adjust_max_liftime(info, &feeder);
+				}
+			})?;
+
+			Keys::<T>::insert(
+				&collection_id,
+				&key,
+				KeyInfo::new(
+					feeders
+						.try_into()
+						.expect("Size of vec matches size of set at most. qed."),
+					min_feeders.unwrap_or_default(),
+				),
 			);
```

Comment:
Other editor for the same collection can overwrite what other editors has set for the same key